flink-1.17-connector-bigquery/flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/source/reader/deserializer/AvroToRowDataConverters.java

```diff
@@ -0,0 +1,363 @@
+
+import static org.apache.flink.formats.avro.typeutils.AvroSchemaConverter.extractValueTypeToAvroMap;
+
+/** Tool class used to convert from Avro {@link GenericRecord} to {@link RowData}. * */
+@Internal
+public class AvroToRowDataConverters {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AvroToRowDataConverters.class);
+
+    /**
+     * Runtime converter that converts Avro data structures into objects of Flink Table & SQL
+     * internal data structures.
+     */
+    @FunctionalInterface
+    public interface AvroToRowDataConverter extends Serializable {
+        Object convert(Object object);
+    }
+
+    // -------------------------------------------------------------------------------------
+    // Runtime Converters
+    // -------------------------------------------------------------------------------------
+
+    public static AvroToRowDataConverter createRowConverter(RowType rowType) {
+        final AvroToRowDataConverter[] fieldConverters =
+                rowType.getFields().stream()
+                        .map(RowType.RowField::getType)
+                        .map(AvroToRowDataConverters::createNullableConverter)
+                        .toArray(AvroToRowDataConverter[]::new);
+        final int arity = rowType.getFieldCount();
+
+        return avroObject -> {
+            IndexedRecord record = (IndexedRecord) avroObject;
+            GenericRowData row = new GenericRowData(arity);
+            for (int i = 0; i < arity; ++i) {
+                // avro always deserialize successfully even though the type isn't matched
+                // so no need to throw exception about which field can't be deserialized
+                row.setField(i, fieldConverters[i].convert(record.get(i)));
+            }
+            return row;
+        };
+    }
+
+    /** Creates a runtime converter which is null safe. */
+    private static AvroToRowDataConverter createNullableConverter(LogicalType type) {
+        final AvroToRowDataConverter converter = createConverter(type);
+        return avroObject -> {
+            if (avroObject == null) {
+                return null;
+            }
+            return converter.convert(avroObject);
+        };
+    }
+
+    /** Creates a runtime converter which assuming input object is not null. */
+    private static AvroToRowDataConverter createConverter(LogicalType type) {
+        switch (type.getTypeRoot()) {
+            case NULL:
+                return avroObject -> null;
+            case TINYINT:
+                return avroObject -> ((Integer) avroObject).byteValue();
+            case SMALLINT:
+                return avroObject -> ((Integer) avroObject).shortValue();
+            case BOOLEAN: // boolean
+            case INTEGER: // int
+            case INTERVAL_YEAR_MONTH: // long
+            case BIGINT: // long
+            case INTERVAL_DAY_TIME: // long
+            case FLOAT: // float
+            case DOUBLE: // double
```

Comment:
will all of these cases fall through to double?

Reply:
Yes, since all need to be converted to respective datatype as it is. 
`avroObject -> avroObject`